### PR TITLE
fix: start payload builder on already canon head

### PR DESCRIPTION
**Pull request by Rjected**
This was stopping basically all hive tests, we still need to start a payload build process if the head is already canonical. In this case we also have to look up the canonical header.

### CODE REVIEW

File: crates/engine/tree/src/tree/mod.rs [hunk 1]
```diff
@@ -1794,11 +1809,6 @@ where
                 return Ok(TreeOutcome::new(outcome))
             }
 
-            if let Some(attr) = attrs {
-                let updated = self.process_payload_attributes(attr, &tip, state);
-                return Ok(TreeOutcome::new(updated))
-            }
-
```
Comment (mattsse) [inline crates/engine/tree/src/tree/mod.rs:1801]: why is this removed here?

we still need this if the fcu.head is new?

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/engine/tree/src/tree/mod.rs (hunk 1 above)
Comment (Rjected) [reply crates/engine/tree/src/tree/mod.rs:1801]: whoops, artifact from an earlier change, we still need this yeah

File: crates/engine/tree/Cargo.toml [hunk 2]
```diff
@@ -85,3 +85,8 @@ test-utils = [
   "reth-static-file",
   "reth-tracing",
 ]
+optimism = [
+    "reth-primitives/optimism",
+    "reth-provider/optimism",
+    "reth-revm/optimism"
+]
```
Comment (mattsse) [inline crates/engine/tree/Cargo.toml:92]: I really want to avoid this, why do we need this?

File: crates/engine/tree/src/tree/mod.rs [hunk 3]
```diff
@@ -1787,18 +1802,18 @@ where
             let tip = chain_update.tip().header.clone();
             self.on_canonical_chain_update(chain_update);
 
+            if let Some(attr) = attrs {
+                let updated = self.process_payload_attributes(attr, &tip, state);
+                return Ok(TreeOutcome::new(updated))
+            }
```
Comment (mattsse) [inline crates/engine/tree/src/tree/mod.rs:1808]: do we want to move this before the following checks?

File: crates/engine/tree/Cargo.toml (hunk 2 above)
Comment (Rjected) [reply crates/engine/tree/Cargo.toml:92]: optimism tests were stalling, let me double check that it wasn't just a flake

Comment (Rjected) [reply crates/engine/tree/Cargo.toml:92]: ok, turns out it was just a flake, will file an issue for the flake and remove this

File: crates/engine/tree/src/tree/mod.rs (hunk 3 above)
Comment (Rjected) [reply crates/engine/tree/src/tree/mod.rs:1808]: sorry, yeah we definitely need to keep the previous location

**Review by mattsse — APPROVED** [review]
